File: Data Structures & Algorithms/anagram-groups/submission-0.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String, List<String>> anagram = new HashMap<>();
        for (String s: strs) {
            int count[] = new int[26];
            for (char c: s.toCharArray()) {
                count[c-'a']++;
            }
            String key = Arrays.toString(count);
            anagram.putIfAbsent(key, new ArrayList<>());
            anagram.get(key).add(s);
        }
        return new ArrayList<>(anagram.values());
    }
}

File: Data Structures & Algorithms/string-encode-and-decode/submission-0.py
class Solution:

    def encode(self, strs: List[str]) -> str:
        decode = ""
        for i in strs:
            decode += str(len(i)) + "#" + i
        return decode


    def decode(self, s: str) -> List[str]:
        result = []
        count = 0

        while count < len(s):
            index = count
            while s[index] != "#":
                index += 1
            length = int(s[count:index])
            result.append(s[index + 1: index + 1 + length])
            count = index + 1 + length
        return result



